source/extensions/filters/http/fault/fault_filter.cc

```diff
@@ -71,6 +69,29 @@ FaultFilter::~FaultFilter() { ASSERT(!delay_timer_); }
 // if we inject a delay, then we will inject the abort in the delay timer
 // callback.
 Http::FilterHeadersStatus FaultFilter::decodeHeaders(Http::HeaderMap& headers, bool) {
+  // Route-level configuration overrides filter-level configuration
+  // TODO (rshriram): We should not be using Runtimes when reading from route-level
+  // faults until we parameterize the runtime keys for faults based on the route name.
+  // Its okay for the moment, since the only use case of faults with runtimes is
+  // by folks using filter level configuration in the fault filter (mainly Lyft folks).
+  if (callbacks_->route() && callbacks_->route()->routeEntry()) {
+    auto proto_config = callbacks_->route()->routeEntry()->perFilterConfig(
+        Extensions::HttpFilters::HttpFilterNames::get().FAULT);
+    if (!proto_config) {
+      // See if the virtual host has any fault filter config
+      proto_config = callbacks_->route()->routeEntry()->virtualHost().perFilterConfig(
+          Extensions::HttpFilters::HttpFilterNames::get().FAULT);
+    }
+
+    if (proto_config) {
+      const envoy::config::filter::http::fault::v2::HTTPFault per_filter_config =
+          dynamic_cast<const envoy::config::filter::http::fault::v2::HTTPFault&>(*proto_config);
+
+      config_.reset(new FaultFilterConfig(per_filter_config, config_->runtime(),
+                                          config_->statsPrefix(), config_->scope()));
+    }
+  }
+
```

Comment:
i think you can get away with doing this in the `setDecoderFilterCallbacks` method, no? Not sure if the route is known by then...but it seems to work from my tests for the buffer filter.